packages/dcos-integration-test/extra/test_service_discovery.py

```diff
@@ -125,6 +128,173 @@ def _pool_for_mesos_dns():
             assert r_data['my_ip'] == service_points[0].host
 
 
+# There are several combinations of Service Discovery Options we have to try:
+#
+# Containerizers:
+# -Mesos
+# -Docker
+#
+# Network type:
+# -Bridged
+# -Host
+# -Overlay
+# -Overlay with Port Mapping
+#
+# Record type:
+# -Container IP
+# -Agent IP
+# -Auto IP
+#
+# More info can be found here: https://dcos.io/docs/1.8/usage/service-discovery/dns-overview/
+
+DNSHost = 0
+DNSPortMap = 1
+DNSOverlay = 2
+DNSAddresses = collections.namedtuple("DNSAddresses", ["container", "agent", "auto"])
+MarathonAddresses = collections.namedtuple("MarathonAddresses", ["host", "container"])
+
+
+def get_ipv4_addresses(hostname):
+    res = socket.getaddrinfo(hostname, 0, family=socket.AF_INET, type=socket.SOCK_STREAM)
+    return frozenset([sockaddr[0] for (family, type, proto, canonname, sockaddr) in res])
+
+
+def get_dns_addresses_by_app_name(app_name):
+    container_ip_name = '{}.marathon.containerip.dcos.thisdcos.directory'.format(app_name)
+    agent_ip_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(app_name)
+    auto_ip_name = '{}.marathon.autoip.dcos.thisdcos.directory'.format(app_name)
+    container_ips = get_ipv4_addresses(container_ip_name)
+    agent_ips = get_ipv4_addresses(agent_ip_name)
+    auto_ips = get_ipv4_addresses(auto_ip_name)
+    return DNSAddresses(container_ips, agent_ips, auto_ips)
+
+
+def get_marathon_addresses_by_service_points(service_points):
+    marathon_host_addrs = frozenset([point.host for point in service_points])
+    marathon_ip_addrs = frozenset([point.ip for point in service_points])
+    return MarathonAddresses(marathon_host_addrs, marathon_ip_addrs)
+
+
+def replace_marathon_cmd_port(app_definition, port_str):
+    new_app = app_definition.copy()
+
+    cmd_list = new_app['cmd'].split()[:-1]
+    cmd_list.append(port_str)
+    mesos_cmd = ' '.join(cmd_list)
+    new_app['cmd'] = mesos_cmd
+    return new_app
+
+
+def assert_service_discovery(dcos_api_session, app_definition, net_types):
+    """
+    net_types: List of network types
```

Comment:
[minor] - Please expand this command. `List of network types: DNSOverlay, DNSPortMap or DNSHost` applicable for navstar.